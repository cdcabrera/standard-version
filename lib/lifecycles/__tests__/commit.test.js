const commit = require('../commit');

jest.mock('../../run-execFile', () => (...args) => {
  console.info(args);
  return 'mock-exec-file-response';
});

describe('commit', () => {
  const mockInfo = jest.fn();
  let spyInfo;

  beforeEach(() => {
    spyInfo = jest.spyOn(console, 'info').mockImplementation(mockInfo);
  });

  afterEach(() => {
    mockInfo.mockClear();
    spyInfo.mockClear();
  });

  it('should skip returning commit information', async () => {
    await commit({ skip: { commit: true } }, 'lorem ipsum');
    expect(mockInfo.mock.calls).toMatchSnapshot('console');
  });

  it('should skip returning changelog and bump information', async () => {
    await commit({ skip: { changelog: true, bump: true } }, 'lorem ipsum');
    expect(mockInfo.mock.calls).toMatchSnapshot('console');
  });

  it('should run lifecycles returning commit information', async () => {
    await commit(
      {
        bumpFiles: [],
        skip: {},
        scripts: {
          prerelease: 'echo "prerelease lifecycle mock"',
          precommit: 'echo "precommit lifecycle mock"',
          postcommit: 'echo "postcommit lifecycle mock"'
        }
      },
      'lorem ipsum'
    );

    expect(mockInfo.mock.calls).toMatchSnapshot('console');
  });
});
