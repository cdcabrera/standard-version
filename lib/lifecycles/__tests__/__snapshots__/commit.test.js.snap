// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`commit should run lifecycles returning commit information: console 1`] = `
Array [
  Array [
    "<green>✔</green> Running lifecycle script \\"<bold>precommit</bold>\\"",
  ],
  Array [
    "<blue>ℹ</blue> - execute command: \\"<bold>echo \\"precommit lifecycle mock\\"</bold>\\"",
  ],
  Array [
    "<green>✔</green> committing <bold>undefined</bold>",
  ],
  Array [
    Array [
      Object {
        "bumpFiles": Array [],
        "releaseCommitMessageFormat": "precommit lifecycle mock
",
        "scripts": Object {
          "postcommit": "echo \\"postcommit lifecycle mock\\"",
          "precommit": "echo \\"precommit lifecycle mock\\"",
          "prerelease": "echo \\"prerelease lifecycle mock\\"",
        },
        "skip": Object {},
      },
      "git",
      Array [
        "add",
        undefined,
      ],
    ],
  ],
  Array [
    Array [
      Object {
        "bumpFiles": Array [],
        "releaseCommitMessageFormat": "precommit lifecycle mock
",
        "scripts": Object {
          "postcommit": "echo \\"postcommit lifecycle mock\\"",
          "precommit": "echo \\"precommit lifecycle mock\\"",
          "prerelease": "echo \\"prerelease lifecycle mock\\"",
        },
        "skip": Object {},
      },
      "git",
      Array [
        "commit",
        undefined,
        "-m",
        "precommit lifecycle mock
",
      ],
    ],
  ],
  Array [
    "<green>✔</green> Running lifecycle script \\"<bold>postcommit</bold>\\"",
  ],
  Array [
    "<blue>ℹ</blue> - execute command: \\"<bold>echo \\"postcommit lifecycle mock\\"</bold>\\"",
  ],
]
`;

exports[`commit should skip returning changelog and bump information: console 1`] = `
Array [
  Array [
    "<green>✔</green> committing %s",
  ],
]
`;

exports[`commit should skip returning commit information: console 1`] = `Array []`;
